### PR TITLE
edit questions

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#50724](https://edu-redmine.sun-asterisk.vn/issues/50724)

## WHAT (optional)
- Edit question and answer same time

## HOW
- create action update

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Review by ductvh-3178 — APPROVED** [review]

File: app/controllers/admin/questions_controller.rb [hunk 1]
```diff
@@ -44,4 +66,17 @@ def at_least_a_true_answer
...
+  def check_at_least_a_true_answer flag
+    return if flag
+
+    flash[:danger] = t ".must_a_true_answer"
+    redirect_to new_admin_subject_question_path
+  end
+
+  def update_false_answer
+    @question.answers.each do |i|
+      i.update_attribute :is_correct, :false_answer
```
Comment (ghost) [inline app/controllers/admin/questions_controller.rb]: tìm hiểu update_attribute có raise exception k

File: app/controllers/admin/questions_controller.rb [hunk 2]
```diff
@@ -21,6 +21,21 @@ def create
...
+  def update
+    ActiveRecord::Base.transaction do
+      update_false_answer
+      @question.update!(question_params)
+      flag = at_least_a_true_answer.call(@question.answers)
+      raise StandardError unless flag
+
+      flash[:success] = t ".update_successed"
+      redirect_to subject_path(@subject.id)
+    end
```
Comment (ghost) [inline app/controllers/admin/questions_controller.rb]:
```suggestion
    end
    flash[:success] = t ".update_successed"
    redirect_to subject_path(@subject.id)
```

File: app/views/admin/questions/_answer_fields.html.erb [hunk 3]
```diff
@@ -6,8 +6,9 @@
     <span><%= t(".if_true") %></span>
   </div>
   <div class="float-right">
-    <%= link_to_remove_association f do %>
-      <i class="fa-solid fa-trash text-danger"></i>
+    <%= link_to_remove_association f ,class: "dynamic" do %>
```
Comment (ghost) [inline app/views/admin/questions/_answer_fields.html.erb]:
```suggestion
    <%= link_to_remove_association f, class: "dynamic" do %>
```

File: app/views/admin/questions/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      <%= form_for @question, url: admin_subject_question_path do |f| %>
+        <div class="form-group">
+          <%= render "shared/error_messages", object: f.object %>
+          <%= f.label t(".question_name") %>
+          <span class="text-danger"> * </span>
+          <%= f.text_field :content, class: "form-control" %>
+        </div>
+        <div class="form-group">
+          <%= f.fields_for :answers do |t| %>
+            <%= render "answer_edit", :f => t %>
```
Comment (ghost) [inline app/views/admin/questions/edit.html.erb]:
```suggestion
            <%= render "answer_edit", f: t %>
```

File: app/views/subjects/show.html.erb [hunk 5]
```diff
@@ -19,7 +19,7 @@
           <% @questions.each do |i| %>
             <li class="list-group-item">
               <div class="float-left">
-                <%= link_to i.content, edit_admin_subject_question_path(i) %>
+                <%= link_to i.content, edit_admin_subject_question_path(i.subject_id,i.id) %>
```
Comment (ghost) [inline app/views/subjects/show.html.erb]:
```suggestion
                <%= link_to i.content, edit_admin_subject_question_path(i.subject_id, i.id) %>
```

**Pull request merged**